### PR TITLE
don't use /tmp/kafka-streams as state.dir in microservices tests

### PR DESCRIPTION
**Pull request by dguy**
Change the microservices tests etc to not use /tmp/kafka-streams.
Increased timeout in `EmailServiceTest` to 30 seconds
Reduce number of consumer-offset partitions in `EmbeddedKafka` as 50 is overkill for tests.

### CODE REVIEW

**Comment by dguy** [comment]
`EmailServiceTest` failed. I couldn't reproduce it, but i've tried to change the things that could've made it fail (was not possible from the test logs to tell).

File: src/main/java/io/confluent/examples/streams/microservices/EmailService.java [hunk 1]
```diff
@@ -35,53 +37,54 @@
...
-      .with(ORDERS.keySerde(), ORDERS.valueSerde(), PAYMENTS.valueSerde());
 
   public EmailService(Emailer emailer) {
     this.emailer = emailer;
   }
 
   @Override
-  public void start(String bootstrapServers) {
-    streams = processStreams(bootstrapServers, "/tmp/kafka-streams");
+  public void start(String bootstrapServers, String stateDir) {
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/microservices/EmailService.java]: nit: add `final` (2x)

File: src/main/java/io/confluent/examples/streams/microservices/EmailService.java [hunk 2]
```diff
@@ -35,53 +37,54 @@
...
 
     //Create the streams/tables for the join
-    KStream<String, Order> orders = builder.stream(ORDERS.keySerde(), ORDERS.valueSerde(), ORDERS.name());
-    KStream<String, Payment> payments = builder.stream(PAYMENTS.keySerde(), PAYMENTS.valueSerde(), PAYMENTS.name());
-    GlobalKTable<Long, Customer> customers = builder.globalTable(CUSTOMERS.keySerde(), CUSTOMERS.valueSerde(), CUSTOMERS.name());
+    KStream<String, Order> orders = builder.stream(ORDERS.name(),
+        Consumed.with(ORDERS.keySerde(), ORDERS.valueSerde()));
+    KStream<String, Payment> payments = builder.stream(PAYMENTS.name(),
+        Consumed.with(PAYMENTS.keySerde(), PAYMENTS.valueSerde()));
+    GlobalKTable<Long, Customer> customers = builder.globalTable(CUSTOMERS.name(),
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/microservices/EmailService.java]: add `final` (also above)

File: src/main/java/io/confluent/examples/streams/microservices/FraudService.java [hunk 3]
```diff
@@ -47,8 +47,8 @@
   private KafkaStreams streams;
 
   @Override
-  public void start(String bootstrapServers) {
-    streams = processStreams(bootstrapServers, "/tmp/kafka-streams");
+  public void start(String bootstrapServers, String stateDir) {
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/microservices/FraudService.java]: nit: `final`

File: src/main/java/io/confluent/examples/streams/microservices/InventoryService.java [hunk 4]
```diff
@@ -48,8 +48,8 @@
   private KafkaStreams streams;
 
   @Override
-  public void start(String bootstrapServers) {
-    streams = processStreams(bootstrapServers, "/tmp/kafka-streams");
+  public void start(String bootstrapServers, String stateDir) {
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/microservices/InventoryService.java]: `final`

File: src/main/java/io/confluent/examples/streams/microservices/OrderDetailsService.java [hunk 5]
```diff
@@ -53,7 +53,7 @@
   private volatile boolean running;
 
   @Override
-  public void start(String bootstrapServers) {
+  public void start(String bootstrapServers, String stateDir) {
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/microservices/OrderDetailsService.java]: `final`

File: src/main/java/io/confluent/examples/streams/microservices/OrdersService.java [hunk 6]
```diff
@@ -307,7 +307,7 @@ public void submitOrder(final OrderBean order,
   }
 
   @Override
-  public void start(String bootstrapServers) {
+  public void start(String bootstrapServers, String stateDir) {
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/microservices/OrdersService.java]: `final`

File: src/main/java/io/confluent/examples/streams/microservices/ValidationsAggregatorService.java [hunk 7]
```diff
@@ -56,8 +56,8 @@
   private KafkaStreams streams;
 
   @Override
-  public void start(String bootstrapServers) {
-    streams = aggregateOrderValidations(bootstrapServers, "/tmp/kafka-streams");
+  public void start(String bootstrapServers, String stateDir) {
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/microservices/ValidationsAggregatorService.java]: `final`

**Pull request merged**